### PR TITLE
TI-2869 - Refactor audit user logging

### PR DESCRIPTION
**Pull request by stewartm83**
Changes:
1) Refactor the client ip address capturing code to favour the server's hostname. We have not been able to log users ip addresses correctly, code returns 127.0.0.1 addresses which are not userful.
2) Refactor the auditUserLogAction method to take in extra parameters userId and username, for the cases where we do not have the logged in user's details. For example when resetting password and trying to log in.
3) Make reset password finish take in an extra config parameter for checking whether we can disable users from resusing old passwords.
 
Please review

### CODE REVIEW

File: imqsauth/net_tools.go [hunk 1]
```diff
@@ -84,25 +36,23 @@ func getIPAddress(r *http.Request) string {
...
+	if serverAddress == "" {
+		serverAddress = getIPAddress(req.http)
 	}
-
-	var loggedInUserId authaus.UserId
-
 	contextDetails := ContextDetails{
-		Service: "Auth",
-		Origin:  getIPAddress(req.http),
+		Service:  "auth",
```
Comment (RoanBrand) [inline imqsauth/net_tools.go]: You are changing to lowercase here. Don't know what the effect might be.

File: imqsauth/http.go [hunk 2]
```diff
@@ -516,7 +516,7 @@ func httpHandlerLogin(central *ImqsCentral, w http.ResponseWriter, r *httpReques
 		authaus.HttpSendTxt(w, http.StatusForbidden, err.Error())
 	} else {
 		r.token = token
-		auditUserLogAction(central, r, "User Profile: "+token.Identity, authaus.AuditActionUserAuthentication)
+		auditUserLogAction(central, r, token.UserId, token.Username, "User Profile: "+token.Identity, authaus.AuditActionUserAuthentication)
```
Comment (RoanBrand) [inline imqsauth/http.go]: Maybe pass in `&authaus.AuthUser` instead of both user id and user name?

File: imqsauth/net_tools.go [hunk 3]
```diff
@@ -111,8 +61,10 @@ func auditUserLogAction(central *ImqsCentral, req *httpRequest, item string, act
 	}
 
 	if central.Central.Auditor != nil {
-		if loggedInUser, err := central.Central.GetUserFromUserId(authaus.UserId(loggedInUserId)); err == nil {
-			central.Central.Auditor.AuditUserAction(loggedInUser.Username, item, string(contextData), actionType)
+		if user, err := central.Central.GetUserFromUserId(authaus.UserId(actorUserId)); err == nil {
+			central.Central.Auditor.AuditUserAction(user.Username, description, string(contextData), actionType)
+		} else {
+			central.Central.Auditor.AuditUserAction(username, description, string(contextData), actionType)
```
Comment (RoanBrand) [inline imqsauth/net_tools.go]: Won't `user.Username` and `username` always be the same?

**Comment by bmharper** [comment]
Can you add a detailed commit message explaining the changes here? It's hard to know, for example, why the removal of the "isPrivateIP" stuff, etc. Commits on the security stuff must always include details explanations of why we're doing things.

**Pull request merged**